lib/router/router.js

```diff
@@ -15,6 +19,10 @@ export default class Router extends EventEmitter {
 
     // set up the component cache (by route keys)
     this.components = { [this.route]: { Component, err } }
+    // contain a map of response of prefetched routes
+    this.prefetchedRoutes = {}
+    this.prefetchingLockManager = new LockManager(2)
+    this.prefetchingRoutes = {}
```

Comment:
Actually, we can't simplify. But we could move the isolate the fetching logic and move it to some other module. Since we are fetching in parallel, we need to keep these states.

Do you wanna move the fetching logic to somewhere else?